src/icalendar/cal.py

```diff
@@ -612,12 +607,8 @@ def to_tz(self):
                 tzname = component['TZNAME'].encode('ascii', 'replace')
                 tzname = self._make_unique_tzname(tzname, tznames)
             except KeyError:
-                tzname = '{}_{}_{}_{}'.format(
-                    zone,
-                    component['DTSTART'].to_ical().decode('utf-8'),
-                    component['TZOFFSETFROM'].to_ical(),  # for whatever reason this is str/unicode
-                    component['TZOFFSETTO'].to_ical(),  # for whatever reason this is str/unicode
-                )
+                # for whatever reason this is str/unicode
+                tzname = f"{zone}_{component['DTSTART'].to_ical().decode('utf-8')}_{component['TZOFFSETFROM'].to_ical()}_{component['TZOFFSETTO'].to_ical()}"  
```

Comment:
This line got quite long.

Reply:
Ok, @angatha I shall split it into multiple lines for readability.